Frameworks/CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -125,39 +113,18 @@ CF_INLINE void __CFTimeZoneUnlockCompatibilityMapping(void) {
  * (Aleksey Dukhnyakov)
  */
 static CFMutableArrayRef __CFCopyWindowsTimeZoneList() {
-// HACKHACK: can't access registry in an appcontainer. Use something different for time zone.
-/*
-    CFMutableArrayRef result = NULL;
-    HKEY hkResult;
-    TCHAR lpName[MAX_PATH+1];
-    DWORD dwIndex, retCode;
-
-    if (RegOpenKey(HKEY_LOCAL_MACHINE,_T(TZZONEINFO),&hkResult) !=
-        ERROR_SUCCESS )
-        return NULL;
-
-    result = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
-    for (dwIndex=0; (retCode = RegEnumKey(hkResult,dwIndex,lpName,MAX_PATH)) != ERROR_NO_MORE_ITEMS ; dwIndex++) {
-        if (retCode != ERROR_SUCCESS) {
-            RegCloseKey(hkResult);
-            CFRelease(result);
-            return NULL;
-        } else {
-#if defined(UNICODE)
-        CFStringRef string = CFStringCreateWithBytes(kCFAllocatorSystemDefault, (const UInt8 *)lpName, (_tcslen(lpName) * sizeof(UniChar)), kCFStringEncodingUnicode, false);
-#else
-        CFStringRef string = CFStringCreateWithBytes(kCFAllocatorSystemDefault, lpName, _tcslen(lpName), CFStringGetSystemEncoding(), false);
-#endif
+    // WINOBJC: can't access registry in an appcontainer. Use list of dynamic time zones instead.
+    CFMutableArrayRef result = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
+    DWORD dwIndex;
+    DYNAMIC_TIME_ZONE_INFORMATION info;
+    
+    for (dwIndex=0; EnumDynamicTimeZoneInformation(dwIndex, &info) == ERROR_SUCCESS ; dwIndex++) {
+        CFStringRef string = CFStringCreateWithBytes(kCFAllocatorSystemDefault, (const UInt8 *)info.StandardName, (wcsnlen_s(info.StandardName, 32) * sizeof(UniChar)), kCFStringEncodingUnicode, false);
```

Comment:
>CFStringCreateWithBytes [](start = 29, length = 23)

Since we already have a buffer of unicode characters in the appropriate format, we should use `CFStringCreateWithCharacters` - it avoids the sizing dance with `sizeof(UniChar)`